atlasexec/atlas_test.go

```diff
@@ -420,6 +420,130 @@ func TestMigrateLintWithLogin(t *testing.T) {
 	})
 }
 
+func TestMigratePush(t *testing.T) {
+	type (
+		graphQLQuery struct {
+			Query     string          `json:"query"`
+			Variables json.RawMessage `json:"variables"`
+			PushDir   *struct {
+				Input struct {
+					Slug   string `json:"slug"`
+					Tag    string `json:"tag"`
+					Driver string `json:"driver"`
+					Dir    string `json:"dir"`
+				} `json:"input"`
+			}
+			SyncDir *struct {
+				Input struct {
+					Slug    string                `json:"slug"`
+					Driver  string                `json:"driver"`
+					Dir     string                `json:"dir"`
+					Context *atlasexec.RunContext `json:"context"`
+				} `json:"input"`
+			}
+		}
+		httpTest struct {
+			payloads []graphQLQuery
+			srv      *httptest.Server
+		}
+	)
+	token := "123456789"
+	newHttpTest := func() (*httpTest, string) {
+		tt := &httpTest{}
+		handler := func() http.HandlerFunc {
+			return func(w http.ResponseWriter, r *http.Request) {
+				require.Equal(t, "Bearer "+token, r.Header.Get("Authorization"))
+				var query graphQLQuery
+				require.NoError(t, json.NewDecoder(r.Body).Decode(&query))
+				tt.payloads = append(tt.payloads, query)
+				if strings.Contains(query.Query, "pushDir") {
+					fmt.Fprint(w, `{"data":{"pushDir":{"url":"https://some-org.atlasgo.cloud/dirs/314159/tags/12345"}}}`)
+				}
+				if strings.Contains(query.Query, "syncDir") {
+					fmt.Fprint(w, `{"data":{"syncDir":{"url":"https://some-org.atlasgo.cloud/dirs/314159/tags/12345"}}}`)
+				}
```

Comment:
If you create a server per test, you can mock the body in the creation. e.g., 
```
newServer(`{"data": {"syncDir": "value"}}`)
```

Reply:
I think it's best to keep it this way, if more tests are added in the future for the same use-case (more tests that need 'syncDir' for example, the response is already built-in and working)